apps/angular/src/app/features/anime/anime-details/anime-details.component.ts

```diff
@@ -0,0 +1,72 @@
+import { ChangeDetectionStrategy, Component, inject } from '@angular/core';
+import { MatDialog } from '@angular/material/dialog';
+import { ActivatedRoute, Router } from '@angular/router';
+import { EMPTY, Observable, map, switchMap } from 'rxjs';
+import { DomSanitizer, SafeResourceUrl } from '@angular/platform-browser';
+import { AnimeService } from '@js-camp/angular/core/services/anime-service';
+import { AnimeDetails } from '@js-camp/core/models/anime/anime-details';
+import { catchAppError } from '@js-camp/angular/core/utils/rxjs/catch-app.error';
+import { AppError } from '@js-camp/core/models/app-error';
+import { NotFoundError } from '@js-camp/core/models/not-found-error';
+
+import { AnimeCoverModalComponent, AnimeCoverModalParameters } from './anime-cover-modal/anime-cover-modal.component';
+
+/** Anime details component. */
+@Component({
+	selector: 'camp-anime-details',
+	templateUrl: './anime-details.component.html',
+	styleUrls: ['./anime-details.component.css'],
+	changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class AnimeDetailsComponent {
+
+	private readonly animeService = inject(AnimeService);
+
+	private readonly activatedRoute = inject(ActivatedRoute);
+
+	private readonly router = inject(Router);
+
+	private readonly dialog = inject(MatDialog);
+
+	private readonly sanitizer = inject(DomSanitizer);
+
+	/** Anime details stream. */
+	protected readonly animeDetails$: Observable<AnimeDetails>;
+
+	public constructor() {
+		this.animeDetails$ = this.activatedRoute.params.pipe(
+			map(({ id }) => Number(id)),
+			switchMap(id => this.animeService.getAnimeById(id)),
+			catchAppError(appError => this.catchAnimeGettingError(appError)),
+		);
+	}
+
+	/**
+	 * Open full size anime image.
+	 * @param imageUrl Image url.
+	 */
+	protected openFullSizeImage(imageUrl: string): void {
+		const modalParameters: AnimeCoverModalParameters = { imageUrl };
+
+		this.dialog.open(AnimeCoverModalComponent, { data: modalParameters });
```

Comment:
You can improve type safety of this code and get hints from IDE about dialog params:
```suggestion
		this.dialog.open<AnimeCoverModalComponent, AnimeCoverModalParameters, void>(
			AnimeCoverModalComponent, { data: modalParameters }
		);
```

`open` method has three generic type params:
- `T` - dialog component (AnimeCoverModalComponent in your case)
- `D` - dialog options, i.e. properties provided in 'data' option (AnimeCoverModalParameters in your case)
- `R` - dialog result, i.e. 'afterDialogClosed()' return type ('void' in your case as dialog doesn't return anything)

Please, prefer to declare these types everywhere when you have a dialog. Otherwise you will end up with 'any'.